dask/dataframe/tests/test_multi.py

```diff
@@ -556,3 +556,22 @@ def test_cheap_single_partition_merge():
 
     list_eq(aa.merge(bb, on='x', how='inner'),
             a.merge(b, on='x', how='inner'))
+
+
+def test_merge_maintains_columns():
```

Comment:
I spent some time trying to get this to work, and did add parametrization to the test for that case, but it turns out that `merge` will call `from_pandas` on its arguments, and `from_pandas` doesn't work with an empty DataFrame. I could take that on but I don't think that would make sense for _this_ PR?